### PR TITLE
[timeseries] Fix NaT timestamp values during resampling

### LINKED ISSUE #4917: [BUG] Getting NaT (Missing timestamp) when trying to convert frequency of data set
**Bug Report Checklist**
I am trying to recreate the advanced timeseries tutorial, I am doing some demand forecast with missing values. 

I load my data into a `TimeSeriesDataFrame` with columns "id", "date_actual" (daily frequency) and "value" (my target) then I am trying to convert to daily frequency and fill in missing values as explained in the website (https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#handling-irregular-data-and-missing-values) but instead I get NaT in the timestamp but with some values assigned , I do get NaN in some columns though, not sure what's going on: 

**Screenshots / Logs**

Image

**Installed Versions**

<details>

INSTALLED VERSIONS
------------------
date                   : 2025-02-21
time                   : 12:19:09.945868
python                 : 3.12.0.final.0
OS                     : Darwin
OS-release             : 24.3.0
Version                : Darwin Kernel Version 24.3.0: Thu Jan  2 20:24:23 PST 2025; root:xnu-11215.81.4~3/RELEASE_ARM64_T8122
machine                : arm64
processor              : arm
num_cores              : 8
cpu_ram_mb             : 8192.0
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 60153

accelerate             : 0.34.2
autogluon              : 1.2
autogluon.common       : 1.2
autogluon.core         : 1.2
autogluon.features     : 1.2
autogluon.multimodal   : 1.2
autogluon.tabular      : 1.2
autogluon.timeseries   : 1.2
boto3                  : 1.36.25
catboost               : 1.2.7
coreforecast           : 0.0.12
defusedxml             : 0.7.1
einops                 : 0.8.1
evaluate               : 0.4.3
fastai                 : 2.7.18
fugue                  : 0.9.1
gluonts                : 0.16.0
huggingface-hub        : 0.29.1
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.5
joblib                 : 1.4.2
jsonschema             : 4.21.1
lightgbm               : 4.5.0
lightning              : 2.5.0.post0
matplotlib             : 3.10.0
mlforecast             : 0.13.4
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.8.1
numpy                  : 1.26.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : None
optimum-intel          : None
orjson                 : 3.10.15
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 11.1.0
psutil                 : 6.1.1
pyarrow                : 19.0.1
pytesseract            : 0.3.10
pytorch-lightning      : 2.5.0.post0
pytorch-metric-learning: 2.3.0
ray                    : 2.39.0
requests               : 2.32.3
scikit-image           : 0.24.0
scikit-learn           : 1.5.2
scikit-learn-intelex   : None
scipy                  : 1.15.2
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.7.5
statsforecast          : 1.7.8
tabpfn                 : None
tensorboard            : 2.19.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.5.1
torchmetrics           : 1.2.1
torchvision            : 0.20.1
tqdm                   : 4.67.1
transformers           : 4.49.0
utilsforecast          : 0.2.4
vowpalwabbit           : None
xgboost                : 2.1.4

</details>

### PR DESCRIPTION
**Pull request by shchur**
*Issue #, if available:* Fixes #4917

*Description of changes:*
- Ensure that the `timestamp` index level has `datetime64[ns]` dtype before resampling. Other dtypes (e.g., `datetime64[ms]` may result in NaT values appearing in the index).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py [hunk 1]
```diff
@@ -1029,7 +1029,11 @@ def resample_chunk(chunk: Iterable[Tuple[str, pd.DataFrame]]) -> pd.DataFrame:
 
         # Resampling time for 1 item < overhead time for a single parallel job. Therefore, we group items into chunks
         # so that the speedup from parallelization isn't dominated by the communication costs.
-        chunks = split_into_chunks(pd.DataFrame(self).groupby(level=ITEMID, sort=False), chunk_size)
+        df = pd.DataFrame(self)
+        # Make sure that timestamp index has dtype 'datetime64[ns]', otherwise index may contain NaT values.
+        # See https://github.com/autogluon/autogluon/issues/4917
+        df.index = df.index.set_levels(df.index.levels[1].astype("datetime64[ns]"), level=TIMESTAMP)
```
Comment (shchur) [inline timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py:1035]: @abdulfatir One possible much cleaner alternative is to ensure that the `timestamp` index level has the `datetime64[ns]` dtype inside `TimeSeriesDataFrame.__init__`.

However, this has a potential downside: `datetime64[ns]` dtype can support the max value of `2262-04-11`, which can be too low in some cases (specifically, for the notorious `m4_yearly` dataset that was constructed by time travelers and contains time series going way beyond year 2300).

Forcing conversion to `datetime64[ns]` only inside the `convert_frequency` method seems like a reasonable comporomise.

Comment (abdulfatir) [reply timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py:1035]: I would not enforce `ns` globally due to the OOB issue you mentioned. That issue is almost paralyzing and there's no way to circumvent it. `convert_frequency` likely has a strictly smaller user base compared to the TDSF, so it's best to keep this logic here. Also, this is something pandas should fix on their side. Should we open an issue there?

**Review by abdulfatir — APPROVED** [review]

File: timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py (hunk 1 above)
Comment (shchur) [reply timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py:1035]: Good point, I've opened #61342
